ui/src/preferences/about-system/UpdatePreferences.tsx

```diff
@@ -1,81 +1,23 @@
 import { Vat } from '@urbit/api';
 import _ from 'lodash';
-import React, { ChangeEvent, FormEvent, useCallback, useEffect, useState } from 'react';
-import { Button } from '../../components/Button';
-import { Setting } from '../../components/Setting';
-import { Spinner } from '../../components/Spinner';
-import { useAsyncCall } from '../../logic/useAsyncCall';
-import useKilnState from '../../state/kiln';
-
-interface UpdatePreferencesProps {
-  base: Vat | undefined;
-}
-
-export const UpdatePreferences = ({ base }: UpdatePreferencesProps) => {
-  const { changeOTASource, toggleOTAs } = useKilnState((s) =>
-    _.pick(s, ['toggleOTAs', 'changeOTASource'])
-  );
-  const otasEnabled = base && !(base.arak?.rail?.paused ?? true);
-  const otaSource = base && base.arak.rail?.ship;
-
-  const toggleBase = useCallback((on: boolean) => toggleOTAs('base', on), [toggleOTAs]);
-
-  const [source, setSource] = useState('');
-  const sourceDirty = source !== otaSource;
-  const { status: sourceStatus, call: setOTA } = useAsyncCall(changeOTASource);
-
-  useEffect(() => {
-    if (otaSource) {
-      setSource(otaSource);
-    }
-  }, [otaSource]);
-
-  const handleSourceChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
-    const { target } = e;
-    const value = target.value.trim();
-    setSource(value.startsWith('~') ? value : `~${value}`);
-  }, []);
-
-  const onSubmit = useCallback(
-    (e: FormEvent<HTMLFormElement>) => {
-      e.preventDefault();
-      setOTA(source);
-    },
-    [source]
-  );
-
+import React from 'react';
+import SourceSetter from '../../components/SourceSetter';
+import useKilnState, { usePike } from '../../state/kiln';
+
+export const UpdatePreferences = () => {
+  const desk = 'base';
+  const appName = 'your Urbit';
+  const pike = usePike(desk);
+  const srcShip = pike?.sync?.ship;
+  const { toggleInstall } = useKilnState();
+  
   return (
-    <div className="inner-section space-y-8 relative">
```

Comment:
This should wrap around the system updates source setter. Also not sure about the title changing here @urcades

Reply:
Good call, will make the change

Reply:
Added to both instances of `SourceSetter`, looks much better:

![image](https://user-images.githubusercontent.com/16504501/205400919-03a5ce38-5cde-4e8d-860c-f27e000831e1.png)

![image](https://user-images.githubusercontent.com/16504501/205400938-4c0eb6d3-5fe2-4800-b096-41293a6df6fe.png)